USER_GUIDE.md

```diff
@@ -249,4 +249,28 @@ console.log('Deleting all PITs:');
 var response = await client.deleteAllPits();
 
 console.log(response.body);
+```
+
+## Mark connection as `dead`
```

Comment:
I'd explain what the purpose of this call is, and ensure it fails when called twice.

Reply:
By fail you mean a runtime error should be thrown when `markDead` is called on the same connection?

Reply:
I think so. Unless there's a reason not to.

I still don't quite get how and when this API is used?

Reply:
These methods are probably syntactical sugar for [set status](https://github.com/opensearch-project/opensearch-js/blob/790f610680450a2f8ea92e1985ac8afa7255d579/lib/Connection.js#L202-L205) method (There only 2 valid connection Statuses: DEAD or ALIVE). I still don't understand why it must throw an error when you set status to DEAD twice though. Wouldn't that also introduce new business logic that will cause trouble for users who are marking a connection as DEAD again?

Reply:
Also since we have `set status` in Connection class, it makes more sense to move `markDead` and `markAlive` from ConnectionPool into Connection since you can change the status of the connection directly anyway. That is instead of `client.connectionPool.markAlive(connection)`, we will do `connection.markAlive()`. A lot more convenient and less confusing because it has nothing to do with ConnectionPool.

Reply:
Sounds like an RFC to me
Should we implement this?

Reply:
I think we should move it to `Connection` for the reason mentioned above